AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/AWSModelReconciliationQueue.swift

```diff
@@ -262,7 +264,10 @@ extension AWSModelReconciliationQueue {
         return errorTypeValue
     }
 
-    private func isUnauthorizedError(_ error: Error?) -> Bool {
+    private func isUnauthorizedError(errorDescription: String, _ error: Error?) -> Bool {
+        if errorDescription.range(of: "Unauthorized", options: .caseInsensitive) != nil {
```

Comment:
Can we use `Error` type instead of using finding presence of string "Unauthorized" in error description?

Reply:
The `Error` type here is the underlying error coming from `APIError.operationError` for subscription messages that contain a valid GraphQL response with the errors in the response payload, containing the unauthorized error. This check against the message is for another scenario at the top level websocket connection that can receive an unauthorized message. 

Some additional details around this.. The AppSync RT client returns the error back to APIPlugin, which then sets the message with operationError that represents "Unauthorized". This was done in this PR https://github.com/aws-amplify/amplify-ios/pull/1994 In that PR we also exposed APIError.isUnauthorized to make it easy for the developer using APIPlugin directly to used. We couldn't use that extension method here in DataStore since APIPlugin is injected at runtime, it doesn't have a dependency on AWSAPIPlugin, making this check the check that we need to do for the websocket connection failures. Alternatively, instead of checking against the string, i think the ideal solution is to check that the underlying error contains AuthError. This just means that APIPlugin needs to fabricate the return object to be an APIError with an underlying AuthError, which isn't hard to do.